src/main/proto/th2_grpc_common/common.proto

```diff
@@ -200,8 +200,27 @@ message Checkpoint {
   string id = 1;
   map<string, DirectionCheckpoint> session_alias_to_direction_checkpoint = 2;
 
+  message CheckpointData {
+    int64 sequence = 1; // Message sequence number in session
+    google.protobuf.Timestamp timestamp = 2; // Message creation timestamp
+  }
+
   message DirectionCheckpoint {
-    map<int32, int64> direction_to_sequence = 1;
+    message DirectionToSequence {
+      option deprecated = true;
+      map<int32, int64> direction_to_sequence = 1;
+    }
+    message DirectionToCheckpointData {
+      map<int32, CheckpointData> direction_to_checkpoint_data = 1;
+    }
```

Comment:
We cannot apply those changes because they break backward compatibilities. We must leave the direction_to_sequence as it was